ckanext/harvest/harvesters/base.py

```diff
@@ -287,7 +287,7 @@ def _create_or_update_package(self, package_dict, harvest_object):
             try:
                 log.debug('Attempting adding unique Identifier to name %s' % package_dict['name'])
                 model.Session.rollback()
-                package_dict['name'] = '%s-%s' % (package_dict['name'], str(uuid.uuid4())[:5])
+                package_dict['name'] = '%s%s' % (package_dict['name'], str(uuid.uuid4())[:5])
```

Comment:
thinking we can leave the dash and just shortens the rand suffix instead 

If the random suffix still results in a duplicate, is it retried?

With 4 hex characters, that's 16^4 possibilities, <1/65000 chance there will be a conflict.

Reply:
I don't think it does, but I agree it would be a very short conflict

Reply:
I agree, I'm willing to live with an error rate below 0.002%...
Going to merge.